tests/test_vqe_vqemcscf.py

```diff
@@ -42,31 +45,52 @@ def test_vqecasci_h2o_2e_2o():
 
 
 def test_ssvqecasci_h2o_2e_2o():
+
+    n_electron = 2
+    n_orbitals = 2
+    excitation_number = 5
+
+    # Generate inital states
+    inital_states = _generate_inital_states(n_electron, n_orbitals, excitation_number)
+    state_2 = inital_states[2]
+    state_3 = inital_states[3]
+
+    # Define spin-adapted linear combinations
+    state_2_spin_adapted = comp_basis_superposition(
+        state_2, state_3, theta=+pi / 4, phi=0.0
+    )
+    state_3_spin_adapted = comp_basis_superposition(
+        state_2, state_3, theta=-pi / 4, phi=0.0
+    )
+
+    # Replace initial states with spin-adapted states. Needed as Ansatz is spin conserving
+    inital_states[2] = state_2_spin_adapted
+    inital_states[3] = state_3_spin_adapted
+
     mc = vqemcscf.VQECASCI(
         mf,
-        2,
-        2,
+        n_electron,
+        n_orbitals,
         optimizer=LBFGS(),
+        inital_states=inital_states,
         ansatz=Ansatz.GateFabric,
         layers=2,
-        excitation_number=3,
+        excitation_number=inital_states,
```

Comment:
You have `excitation_number=inital_states`, but should this actually be `excitation_number=len(initial_states)` or something like that? It feels odd for number to not be an integer.